### DISCUSSION TITLE
How can i change MY_POD_IP via external pod service_name

### DISCUSSION THREAD

**Discussion by sonerakinci** [open-telemetry/community#2024] [Community Discussions]
Hi.
I used before opentelemetry on docker.  It was so easy.  But now, i am trying to run my containers on k8s a lot of bad experience with opentelemetry. 

When i changed the MY_POD_IP via pod service_name the app could not connect.

Error Message:

Error: cannot start pipelines: listen tcp 10.43.220.21:9411: bind: cannot assign requested address
collector server run finished with error: cannot start pipelines: listen tcp 10.43.220.21:9411: bind: cannot assign requested address

docker(working):
`receivers:
  otlp:
    protocols:
      http:
      grpc:

processors:
  # batch metrics before sending to reduce API usage
  batch:

exporters:
  logging:
    loglevel: debug
  prometheus:
    endpoint: "0.0.0.0:8889"
    const_labels:
      label1: spring
  otlp:
    endpoint: "http://tempo:4317"  # tempo container host and port
    tls:
      insecure: true
  loki:
    endpoint: "http://loki:3100/loki/api/v1/push"
  zipkin:
    endpoint: "http://zipkin:9411/api/v2/spans"
    format: proto
  jaeger:
    endpoint: "http://jaeger-service:14250"
    tls:
      insecure: true

extensions:
  # responsible for responding to health check calls on behalf of the collector.
  health_check:
  # fetches the collector’s performance data
  pprof:
  # serves as an http endpoint that provides live debugging data about instrumented components.
  zpages:

service:
  extensions: [health_check, pprof, zpages]
  pipelines:
    # for now we only interested about metrics...
    metrics:
      receivers: [otlp]
      processors: [batch]
      exporters: [prometheus]
    traces:
      receivers: [otlp]
      processors: [batch]
      exporters: [otlp,zipkin,jaeger]  # name here should match the exporter name for tempo which is otlp
    logs:
      receivers: [otlp]
      exporters: [loki,logging]`
  

k8s(not working):
```
config:
  exporters:
    debug: {}
  extensions:
    # The health_check extension is mandatory for this chart.
    # Without the health_check extension the collector will fail the readiness and liveliness probes.
    # The health_check extension can be modified, but should never be removed.
    health_check:
      endpoint: ${env:MY_POD_IP}:13133
    memory_ballast: {}
  processors:
    batch: {}
    # If set to null, will be overridden with values based on k8s resource limits
    memory_limiter: null
  receivers:
    jaeger:
      protocols:
        grpc:
          endpoint: ${env:MY_POD_IP}:14250
        thrift_http:
          endpoint: ${env:MY_POD_IP}:14268
        thrift_compact:
          endpoint: ${env:MY_POD_IP}:6831
    otlp:
      protocols:
        grpc:
          endpoint: ${env:MY_POD_IP}:4317
        http:
          endpoint: ${env:MY_POD_IP}:4318
    prometheus:
      config:
        scrape_configs:
          - job_name: opentelemetry-collector
            scrape_interval: 10s
            static_configs:
              - targets:
                  - ${env:MY_POD_IP}:8888
    zipkin:
      endpoint: ${env:MY_POD_IP}:9411
  service:
    telemetry:
      metrics:
        address: ${env:MY_POD_IP}:8888
    extensions:
      - health_check
      - memory_ballast
    pipelines:
      logs:
        exporters:
          - debug
        processors:
          - memory_limiter
          - batch
        receivers:
          - otlp
      metrics:
        exporters:
          - debug
        processors:
          - memory_limiter
          - batch
        receivers:
          - otlp
          - prometheus
      traces:
        exporters:
          - debug
        processors:
          - memory_limiter
          - batch
        receivers:
          - otlp
          - jaeger
          - zipkin
```

**Comment by danielgblanco** [comment]
Hi @sonerkoylu you will probably find folks able to help you in the `#otel-collector` CNCF Slack channel. See https://cloud-native.slack.com/archives/C01N6P7KR6W
